src/Responder.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <functional>
+#include <sstream>
+#include <map>
+#include <thread>
+#include <iostream>
+extern "C" {
+    #include <czmq.h>
+}
+
+#include "RedisDiscover.h"
+#include "config.h"
+
+class Responder {
+    std::string address_;
+    RedisDiscover redis_discover_;
+    std::map<std::string, std::function <void(const std::stringstream&, std::stringstream&)>> handlers_;
```

Comment:
you should create a typedef for this function

using handler = std::function<...>;